components/cam/src/physics/cam/shoc_intr.F90

```diff
@@ -0,0 +1,1246 @@
+  !                                                                                 !
+  ! =============================================================================== !
+  
+  subroutine shoc_register_e3sm()
+
+#ifdef SHOC_SGS
+    ! Add SHOC fields to pbuf
+    use physics_buffer,  only: pbuf_add_field, dtype_r8, dyn_time_lvls
+    use ppgrid,          only: pver, pverp, pcols  
+    
+    call phys_getopts( eddy_scheme_out                 = eddy_scheme, &
+                       deep_scheme_out                 = deep_scheme, & 
+                       do_tms_out                      = do_tms,      &
+                       history_budget_out              = history_budget, &
+                       history_budget_histfile_num_out = history_budget_histfile_num, &
+                       micro_do_icesupersat_out        = micro_do_icesupersat, &
+                       micro_mg_accre_enhan_fac_out    = micro_mg_accre_enhan_fac)    
+ 
+    cnst_names=(/'TKE   '/)    
+ 
+    ! TKE is prognostic in SHOC and should be advected by dynamics
+    call cnst_add('SHOC_TKE',0._r8,0._r8,0._r8,ixtke,longname='turbulent kinetic energy',cam_outfld=.false.)
+  
+    ! Fields that are not prognostic should be added to PBUF
+    call pbuf_add_field('WTHV', 'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), wthv_idx)
+    call pbuf_add_field('TKH', 'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), tkh_idx) 
+    call pbuf_add_field('TK', 'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), tk_idx) 
+
+    call pbuf_add_field('pblh',       'global', dtype_r8, (/pcols/), pblh_idx)
+    call pbuf_add_field('tke',        'global', dtype_r8, (/pcols, pverp/), tke_idx)
+    call pbuf_add_field('kvh',        'global', dtype_r8, (/pcols, pverp/), kvh_idx)
+    call pbuf_add_field('kvm',        'global', dtype_r8, (/pcols, pverp/), kvm_idx)
+    call pbuf_add_field('tpert',      'global', dtype_r8, (/pcols/), tpert_idx)
+    call pbuf_add_field('AST',        'global', dtype_r8, (/pcols,pver,dyn_time_lvls/),    ast_idx)
+    call pbuf_add_field('AIST',       'global', dtype_r8, (/pcols,pver,dyn_time_lvls/),    aist_idx)
+    call pbuf_add_field('ALST',       'global', dtype_r8, (/pcols,pver,dyn_time_lvls/),    alst_idx)
+    call pbuf_add_field('QIST',       'global', dtype_r8, (/pcols,pver,dyn_time_lvls/),    qist_idx)
+    call pbuf_add_field('QLST',       'global', dtype_r8, (/pcols,pver,dyn_time_lvls/),    qlst_idx)
+    call pbuf_add_field('CONCLD',     'global', dtype_r8, (/pcols,pver,dyn_time_lvls/),    concld_idx)
+    call pbuf_add_field('CLD',        'global', dtype_r8, (/pcols,pver,dyn_time_lvls/),    cld_idx)
+    call pbuf_add_field('FICE',       'physpkg',dtype_r8, (/pcols,pver/), fice_idx)
+    call pbuf_add_field('RAD_CLUBB',  'global', dtype_r8, (/pcols,pver/), radf_idx)
+    call pbuf_add_field('CMELIQ',     'physpkg',dtype_r8, (/pcols,pver/), cmeliq_idx)
+    
+    call pbuf_add_field('vmag_gust',  'global', dtype_r8, (/pcols/),      vmag_gust_idx)
```

Comment:
There are a set of pbuf variables that are added by CLUBB, not listed here.  I assume these are all variables that aren't needed by other modules, or are CLUBB specific and are not need by SHOC?  They are:

```

    call pbuf_add_field('WP2_nadv',        'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), wp2_idx)
    call pbuf_add_field('WP3_nadv',        'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), wp3_idx)
    call pbuf_add_field('WPTHLP_nadv',     'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), wpthlp_idx)
    call pbuf_add_field('WPRTP_nadv',      'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), wprtp_idx)
    call pbuf_add_field('RTPTHLP_nadv',    'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), rtpthlp_idx)
    call pbuf_add_field('RTP2_nadv',       'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), rtp2_idx)
    call pbuf_add_field('THLP2_nadv',      'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), thlp2_idx)
    call pbuf_add_field('UP2_nadv',        'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), up2_idx)
    call pbuf_add_field('VP2_nadv',        'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), vp2_idx)

    call pbuf_add_field('UPWP',       'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), upwp_idx)
    call pbuf_add_field('VPWP',       'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), vpwp_idx)
    call pbuf_add_field('THLM',       'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), thlm_idx)
    call pbuf_add_field('RTM',        'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), rtm_idx)
    call pbuf_add_field('UM',         'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), um_idx)
    call pbuf_add_field('VM',         'global', dtype_r8, (/pcols,pverp,dyn_time_lvls/), vm_idx)
```

Reply:
those are CLUBB's prognostic variables for the higher order moments, which need to be saved from time step to time step... and not used anywhere else in the E3SM code.  SHOC's higher order moments are diagnostic, thus do not need to be saved from time step to time step.

Reply:
Excellent, that's what I thought, but wanted to check.